src/maker.rs

```diff
@@ -141,45 +173,55 @@ impl Maker {
                         position: Position::Buy,
                         ..
                     } => {
-                        // We are buying BTC for DAI
-                        // Given an order rate of: 1:9000
-                        // It is NOT profitable to buy, if the current rate is greater than the order rate.
-                        // 1:8800 -> We give less DAI for getting BTC -> Good.
-                        // 1:9200 -> We have to give more DAI for getting BTC -> Sucks.
-                        if order_rate > current_profitable_rate {
-                            return Ok(TakeRequestDecision::RateNotProfitable);
-                        }
-
-                        let updated_dai_reserved_funds =
-                            self.dai_reserved_funds.clone() + order.quote;
-                        if updated_dai_reserved_funds > self.dai_balance {
-                            return Ok(TakeRequestDecision::InsufficientFunds);
+                        match self.dai_balance.clone() {
+                            Some(dai_balance) => {
+                                // We are buying BTC for DAI
```

Comment:
```suggestion
                        match self.dai_balance {
                            Some(ref dai_balance) => {
                                // We are buying BTC for DAI
```
Clone later :)

Reply:
I did the `ref` and `*` (deref) upon comparison below. Is that more idiomatic than the clone?

Reply:
The `ref` way is not more idiomatic, it is just better because you avoid a `clone`. You use less memory and it's faster because you are not copying data.